### PR TITLE
fix(forge-wizard): project name validation to match forge backend

### PR DESCRIPTION
**Pull request by corinnekrych**
Fixes https://github.com/openshiftio/openshift.io/issues/1156 and https://github.com/openshiftio/openshift.io/issues/1157 and https://github.com/openshiftio/openshift.io/issues/1210

### CODE REVIEW

**Comment by fabric8cd** [comment]
@corinnekrych fabric8/fabric8-ui:SNAPSHOT-PR-2221-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2221-fabric8-ui.badger.fabric8.io

File: src/app/space/forge-wizard/quickstart-pages/step2/project-info-step.component.ts [hunk 1]
```diff
@@ -19,8 +19,14 @@ export class ProjectInfoStepComponent implements OnInit {
   constructor() {}
 
   ngOnInit(): void {
+    // Add validation to fit server side validation
+    this.form.controls['named'].setValidators([Validators.pattern(/^[a-z][a-z0-9\-]*$/),
+      Validators.minLength(2),
+      Validators.maxLength(24)]);
```
Comment (edewit) [inline src/app/space/forge-wizard/quickstart-pages/step2/project-info-step.component.ts]: max length should be 63

**Comment by fabric8cd** [comment]
@corinnekrych fabric8/fabric8-ui:SNAPSHOT-PR-2221-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2221-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@corinnekrych fabric8/fabric8-ui:SNAPSHOT-PR-2221-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2221-fabric8-ui.badger.fabric8.io

**Pull request merged**